// @unlock-protocol/unlock-js/lib/v8/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _abis = _interopRequireDefault(require("../abis"));

var _configureUnlock = _interopRequireDefault(require("./configureUnlock"));

var _createLock = _interopRequireDefault(require("./createLock"));

var _keyManagerOf = _interopRequireDefault(require("./keyManagerOf"));

var _purchaseKey = _interopRequireDefault(require("./purchaseKey"));

var _v = _interopRequireDefault(require("../v7"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const {
  getLock,
  grantKey,
  updateKeyPrice,
  withdrawFromLock,
  initializeTemplate,
  isLockManager,
  isKeyGranter,
  addKeyGranter,
  expireAndRefundFor,
  cancelAndRefund,
  shareKey
} = _v.default;
var _default = {
  version: 'v8',
  Unlock: _abis.default.v8.Unlock,
  PublicLock: _abis.default.v8.PublicLock,
  createLock: _createLock.default,
  getLock,
  initializeTemplate,
  updateKeyPrice,
  purchaseKey: _purchaseKey.default,
  grantKey,
  withdrawFromLock,
  configureUnlock: _configureUnlock.default,
  isLockManager,
  isKeyGranter,
  addKeyGranter,
  expireAndRefundFor,
  cancelAndRefund,
  shareKey,
  keyManagerOf: _keyManagerOf.default
};
exports.default = _default;
