// @unlock-protocol/unlock-js/lib/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ZERO = exports.KEY_ID = exports.UNLIMITED_KEYS_COUNT = exports.ETHERS_MAX_UINT = exports.MAX_UINT = exports.GAS_AMOUNTS = void 0;

var _ethers = require("ethers");

// TODO: gas amounts should be custom per version!
const GAS_AMOUNTS = {
  createLock: 4500000,
  // TODO Make dynamic as later version of the protocol are much cheaper!
  updateKeyPrice: 100000,
  purchaseFor: 500000,
  // purchaseKey in walletService
  purchase: 500000,
  // purchaseKey in walletService
  withdraw: 100000,
  // withdrawFromLock in walletService
  deployContract: 6400000,
  updateKeyPricing: 100000
};
exports.GAS_AMOUNTS = GAS_AMOUNTS;
const MAX_UINT = '115792089237316195423570985008687907853269984665640564039457584007913129639935';
exports.MAX_UINT = MAX_UINT;
const ETHERS_MAX_UINT = _ethers.ethers.constants.MaxUint256;
exports.ETHERS_MAX_UINT = ETHERS_MAX_UINT;
const UNLIMITED_KEYS_COUNT = -1;
exports.UNLIMITED_KEYS_COUNT = UNLIMITED_KEYS_COUNT;

const KEY_ID = (lock, owner) => [lock, owner].join('-');

exports.KEY_ID = KEY_ID;
const ZERO = _ethers.ethers.constants.AddressZero;
exports.ZERO = ZERO;
var _default = {
  MAX_UINT,
  UNLIMITED_KEYS_COUNT,
  GAS_AMOUNTS,
  KEY_ID
};
exports.default = _default;
