// @unlock-protocol/unlock-js/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Web3Service", {
  enumerable: true,
  get: function () {
    return _web3Service.default;
  }
});
Object.defineProperty(exports, "WalletService", {
  enumerable: true,
  get: function () {
    return _walletService.default;
  }
});
Object.defineProperty(exports, "getCurrentProvider", {
  enumerable: true,
  get: function () {
    return _providers.getCurrentProvider;
  }
});
Object.defineProperty(exports, "getWeb3Provider", {
  enumerable: true,
  get: function () {
    return _providers.getWeb3Provider;
  }
});
Object.defineProperty(exports, "deploy", {
  enumerable: true,
  get: function () {
    return _deploy.default;
  }
});
exports.latest = void 0;

var _web3Service = _interopRequireDefault(require("./web3Service"));

var _walletService = _interopRequireDefault(require("./walletService"));

var _providers = require("./providers");

var _deploy = _interopRequireDefault(require("./deploy"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const latest = 'v9';
exports.latest = latest;
